bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,268 @@
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command.toFullString());
+
+        if (command == RefreshType.REFRESH) {
+            refreshData();
+            return;
+        }
+
+        switch (channelUID.getIdWithoutGroup()) {
+            case SETPOINT:
+                handleSetpointCommand(channelUID, command);
+                break;
+            case POWER:
+                handlePowerCommand(channelUID, command);
+                break;
+            case MODE:
+                handleModeCommand(channelUID, command);
+                break;
+            case SENSOR:
+                handleSensorCommand(channelUID, command);
+                break;
+            case REMOTE_LOCK:
+                handleRemoteLockCommand(channelUID, command);
+                break;
+            case TIME:
+                handleSetTimeCommand(channelUID, command);
+                break;
+            default:
+                logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handlePowerCommand(ChannelUID channelUID, Command command) {
+        @Nullable
+        FloureonDevice floureonDevice = this.floureonDevice;
+        if (command instanceof OnOffType && floureonDevice != null) {
+            try {
+                floureonDevice.setPower(command == OnOffType.ON);
+            } catch (Exception e) {
+                logger.warn("Error while setting power of {} to {}: {}", thing.getUID(), command, e.getMessage());
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleModeCommand(ChannelUID channelUID, Command command) {
+        @Nullable
+        FloureonDevice floureonDevice = this.floureonDevice;
+        if (command instanceof StringType && floureonDevice != null) {
+            try {
+                if (MODE_AUTO.equals(command.toFullString())) {
+                    floureonDevice.switchToAuto();
+                } else {
+                    floureonDevice.switchToManual();
+                }
+            } catch (Exception e) {
+                logger.warn("Error while setting power off {} to {}: {}", thing.getUID(), command, e.getMessage());
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleSetpointCommand(ChannelUID channelUID, Command command) {
+        @Nullable
+        FloureonDevice floureonDevice = this.floureonDevice;
+        if (command instanceof QuantityType && floureonDevice != null) {
+            try {
+                @Nullable
+                @SuppressWarnings("unchecked")
+                QuantityType<Temperature> temperatureQuantityType = ((QuantityType<Temperature>) command)
+                        .toUnit(SIUnits.CELSIUS);
```

Comment:
You could specify the wildcard to get rid of the supression.
```suggestion
                QuantityType<?> temperatureQuantityType = ((QuantityType<?>) command).toUnit(SIUnits.CELSIUS);
```

Reply:
Ah, nice! Applied.